### PR TITLE
use defer to unlock mutex

### PR DESCRIPTION
**Pull request by rainrambler**
use defer to unlock mutex (clean up)

Signed-off-by: Anyu Wang <wanganyu@outlook.com>

**- What I did**
source code clean up

**- How I did it**
use defer to unlock mutex

**- How to verify it**
N/A

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by coolljt0725** [comment]
@rainrambler  Hi,  please sign your commit correctly
https://github.com/moby/moby/pull/42543/checks?check_run_id=2869351154

File: pkg/ioutils/bytespipe.go [hunk 1]
```diff
@@ -126,16 +124,15 @@ func (bp *BytesPipe) Close() error {
 // Data could be read only once.
 func (bp *BytesPipe) Read(p []byte) (n int, err error) {
 	bp.mu.Lock()
+	defer bp.mu.Unlock()
 	if bp.bufLen == 0 {
 		if bp.closeErr != nil {
 			err := bp.closeErr
-			bp.mu.Unlock()
 			return 0, err
```
Comment (thaJeztah) [inline pkg/ioutils/bytespipe.go:131]: Now that we use the defer, do we still need the intermediate `err` variable?

```go
return 0, bp.closeErr
```

File: pkg/ioutils/bytespipe.go [hunk 2]
```diff
@@ -126,16 +124,15 @@ func (bp *BytesPipe) Close() error {
...
 		}
 		bp.wait.Wait()
 		if bp.bufLen == 0 && bp.closeErr != nil {
 			err := bp.closeErr
-			bp.mu.Unlock()
 			return 0, err
```
Comment (thaJeztah) [inline pkg/ioutils/bytespipe.go:136]: Same here;

```go
return 0, bp.closeErr
```

**Review by thaJeztah — APPROVED** [review]
LGTM, but left a comment / thought

**Comment by thaJeztah** [comment]
Failure on Windows is unrelated; `TestNetworkDBIslands` - known flaky test (tracked through https://github.com/moby/moby/issues/42459)

**Pull request merged**